snapcraft/tests/test_lifecycle.py

```diff
@@ -1211,6 +1211,67 @@ def test_prime_with_plugin_manifest(self, fake_plugin_manifest):
             os.path.join('prime', 'snap', 'manifest.yaml'),
             FileContains(expected))
 
+    def test_prime_with_image_info_records_manifest(self):
+        self.useFixture(fixtures.EnvironmentVariable(
+            'SNAPCRAFT_BUILD_INFO', '1'))
+        test_image_info = (
+            '{"architecture": "test-architecture", '
+            '"created_at": "test-created-at", '
+            '"fingerprint": "test-fingerprint"}')
+        self.useFixture(fixtures.EnvironmentVariable(
+            'SNAPCRAFT_IMAGE_INFO', test_image_info))
+        self.make_snapcraft_yaml(
+            textwrap.dedent("""\
+                parts:
+                  test-part:
+                    plugin: nil
+                """))
+        lifecycle.execute('prime', self.project_options)
+
+        expected = textwrap.dedent("""\
+            name: test
+            version: 0
+            summary: test
+            description: test
+            confinement: strict
+            grade: stable
+            parts:
+              test-part:
+                build-packages: []
+                installed-packages: []
+                installed-snaps: []
+                plugin: nil
+                prime: []
+                stage: []
+                stage-packages: []
+                uname: Linux test uname 4.10 x86_64
+            architectures: [{}]
+            image-info: {}
```

Comment:
For `image-info` it would be nice to see this actually expanded here (manually), instead of using `format` to get a real feel of how it would look like.

This should also look like yaml

Reply:
It's YAML - curly braces are the one-line syntax for dictionaries, analoguous to square braces for lists.
But I agree after putting it into the test as per your suggestion that it feels inconsistent. I'm trying to find out how to force `yaml.dump` to use multi-line syntax here.

Reply:
I would like to see the yaml here please, the revert moved that all away.

Reply:
Sorry, putting it back...